wordpress-mobile/WordPress-Android
# App Data Coverage: Add more tracking to settings

Parent #16343 

This PR adds more tracking to the following setting views:

- My Profile
- Account Settings
- App Settings
- Privacy Settings

**To test:**
- Launch the app with an account you don't care about changing account name, display name, etc. I created a new account for testing.

**Test: My Profile**
- Navigate to Me -> My Profile
- Edit any value on the view
- ✅  Verify the log contains: `🔵 Tracked: settings_did_change, Properties: {"field_name":FIELD_NAME, "page":"my_profile"} 

**Test: Account Settings**
- Navigate to Me -> Account Settings
-  Edit any value on the view
- ✅  Verify the log contains: `🔵 Tracked: settings_did_change, Properties: {"field_name":FIELD_NAME, "page":"account_settings"} 

**Test: App Settings**
- Navigate to Me -> App Settings
- Tap and change Appearance
- ✅  Verify the log contains: `🔵 Tracked: app_settings_appearance_changed, Properties:  {"style":"dark|light|default"}
- Tap Privacy Settings
- ✅  Verify the log contains: 🔵 Tracked: app_settings_privacy_settings_tapped
-  ✅  Verify the log contains: 🔵 Tracked: privacy_settings_opened
- Go back to Privacy Settings
- Tap Open Device Settings
-  ✅  Verify the log contains: 🔵 Tracked: app_settings_open_device_settings_tapped
- Navigate back to the app using back button
- Tap Optimize Images to open up the next two settings
- Change both Maximum Image Size and Image Quality
- ✅  Verify the log contains: 🔵 Tracked: app_settings_max_image_size_changed AND 🔵 Tracked: app_settings_image_quality_changed
- Toggle Remove location from media 
- ✅  Verify the log contains: 🔵 Tracked: app_settings_remove_location_from_media_changed, Properties: {"enabled":true|false}
- Toggle Optimize videos to open up the next two settings
- ✅  Verify the log contains: 🔵 Tracked: app_settings_video_optimization_changed, Properties: {"enabled":true|false}
- Change Maximum Video Size and Video Quality
- ✅  Verify the log contains: 🔵 Tracked: app_settings_max_video_size_changed  AND 🔵 Tracked: app_settings_video_quality_changed 

**Test: Privacy Settings**
- Navigate to Me -> App Settings -> Privacy Settings
- Toggle Crash Reports
- ✅  Verify the log contains: 🔵 Tracked: privacy_settings_report_crashes_toggled, Properties: {"enabled":true|flase}
 
## Regression Notes
1. Potential unintended areas of impact
None, tracking added

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
